### ISSUE TITLE
Undefined index with zircote/swagger-php annotations in class dockblock

### ISSUE THREAD

**Issue by Biiru** [neos/flow-development-collection#2387]
### Description

I tried to use the package "zircote/swagger-php" (https://github.com/zircote/swagger-php) to generate a swagger documentation from Flow. Method annotations works well but if I use an annotation like `@OA\Info()` or `@OA\Schema()`, Flow throw me this error

```
Notice: Undefined index: filename in /var/www/gabwebapi.alloboissons.ch/Packages/Framework/Neos.Flow/Classes/ObjectManagement/Proxy/Compiler.php line 310
Type: Neos\Flow\Error\Exception
Code: 1
File: Packages/Framework/Neos.Flow/Classes/Error/ErrorHandler.php
Line: 81
Open Data/Logs/Exceptions/20210209132528bbf790.txt for a full stack trace.
Exception Code	1355480641
Exception Type	Neos\Flow\Core\Booting\Exception\SubProcessException
Log Reference	202102091325270c9489
Thrown in File	Packages/Framework/Neos.Flow/Classes/Core/Booting/Scripts.php
Line	696
```
### More info

I already talk about that with @albe on Slack and there is his first impression

> How should be deal with annotations like https://github.com/zircote/swagger-php/blob/320ae56fa49f7a305c31e6e75786ce264711073e/src/Annotations/AbstractAnnotation.php#L31 that contain public properties that are automatically created and not supposed to be marked up in the @XX\YY() annotation docblock? I mean, I'm not even entirely sure why we manually rerender the annotation blocks from the instanciated class in the first place - do we ever add properties to annotations in the proxy? (see https://github.com/neos/flow-development-collection/blob/master/Neos.Flow/Classes/ObjectManagement/Proxy/Compiler.php#L285)

https://neos-project.slack.com/archives/C04V4C6B0/p1612877341056400

### Affected Versions

Flow: 7.0.2

Php: 7.4.3

Swagger-php: 3.1.0

**Comment by albe** [comment]
My quickshot solution would be some rule to ignore properties that start with `_` in the compiler, but ofc that's magic dust we add. Unfortunately, I don't see much other option to detect properties that should not end up in the rendered annotation. Only visiting the highest level of inheritance (only properties of e.g. the `Schema` annotation rather than the inherited `AbstractAnnotation`) is not a good solution, because inheriting properties is generally totally fine. The real issue is that the OA annotations are not just plain POPOs, but contain (publicly accessible !) state (the `Context` class).

The only alternative I see is avoiding the rerendering of docblock annotations alltogether and instead just c&p from the source class - but I'm not sure we might want to actually do something with annotations?

**Comment by EnesE** [comment]
I used Flow 5.3.21 before and everything worked fine. I updated than my packages via composer to Flow 5.3.22 (and 5.3.23).
After the update I had the same issue `Undefined index: filename in [...]`.

I was able to locate the commit that might be causing the error: https://github.com/neos/flow/commit/979dbe117963c7397aa10b89350225d893d54ed6

When debugging I was able to find out that the error was coming from this class (and line): https://github.com/zircote/swagger-php/blob/master/src/Context.php#L24

**Comment by albe** [comment]
I have a potential fix for this in
https://github.com/neos/flow-development-collection/pull/2404/files#diff-75afa6ce93961444495810fd8105e3a7a640f819fe4987c11e33ea6b89ac5502R253-R257 and https://github.com/neos/flow-development-collection/pull/2404/files#diff-c502a29345ccd01f8daf4cd2803ca7800c7b98ff37502fa7cc457f2b573bb37cR225-R230 that would completely skip our re-rendering of docblock annotations for class and method docblocks (we already don't do this for properties). I would extract those changes into an own PR for that.

@kdambekalns do you see any problem with this? My current attempt would be a filter for `_` properties in 5.3+ and switch do c&p docblocks in 7.1

**Comment by albe** [comment]
FYI: A fix (#2406) for this has landed in the latest bugfix releases of all versions 5.3+. I couldn't fully test if swagger-php is now working correctly, so any feedback is highly welcome. If my time allows, I'll try to create a little test-app locally this week that uses a couple of the OA annotations and see if I can get a proper swagger out of it

**Comment by Biiru** [comment]
@albe I just tested your fix and seem be ok for me !
My endpoints works with all the OA Annotations 
Thanks a lot for this fix !
